lib/concurrent/index.js

```diff
@@ -117,14 +122,15 @@ class ArrayBasedExecutor {
         this._stop = false;
     }
 
-    execute() {
+    async execute() {
         const promises = new Array(this._concurrencyLevel);
 
         for (let i = 0; i < this._concurrencyLevel; i++) {
             promises[i] = this._executeOneAtATime(i, 0);
         }
 
-        return Promise.all(promises).then(() => this._result);
+        await Promise.all(promises);
+        return this._result;
```

Comment:
:wrench: This seem unrelated to this commit, and even PR. Either delete this change from this PR or make it a separate commit.